### PR TITLE
ECS simple implementation

### PR DESCRIPTION
**Pull request by GuanHuiLim**
Creation and deletion of entities & components
Registering and usage of systems
Views for iteration through entities with specified components

### CODE REVIEW

File: Engine/src/Engine/ECS/ECS.cpp
Comment (spironan) [inline Engine/src/Engine/ECS/ECS.cpp:2]: remove this file if its not used.

File: Engine/src/Engine/ECS/ECS.h
Comment (spironan) [inline Engine/src/Engine/ECS/ECS.h:4]: same as before, should remove file if not used

File: Engine/src/Engine/ECS/ECS_Test.cpp
Comment (spironan) [inline Engine/src/Engine/ECS/ECS_Test.cpp:2]: once again same as before should remove if unused

File: Engine/src/Engine/ECS/SparseArray.h
Comment (spironan) [inline Engine/src/Engine/ECS/SparseArray.h:15]: Delete away if unused?

File: Engine/src/Engine/ECS/System.h
Comment (spironan) [inline Engine/src/Engine/ECS/System.h:18]: Should Systems be wrapped in engine namespace or not?

File: Engine/src/Engine/ECS/SystemManager.h
Comment (spironan) [inline Engine/src/Engine/ECS/SystemManager.h:22]: similarly for system manager

**Review by spironan — CHANGES_REQUESTED** [review]
Do Get back on the comments before approval

File: Engine/src/Engine/ECS/ECS.cpp
Comment (GuanHuiLim) [reply Engine/src/Engine/ECS/ECS.cpp:2]: It has been removed

File: Engine/src/Engine/ECS/ECS.h
Comment (GuanHuiLim) [reply Engine/src/Engine/ECS/ECS.h:4]: This is the main header file used by others to access the ecs functionality

File: Engine/src/Engine/ECS/ECS_Test.cpp
Comment (GuanHuiLim) [reply Engine/src/Engine/ECS/ECS_Test.cpp]: It has been removed

File: Engine/src/Engine/ECS/SparseArray.h
Comment (GuanHuiLim) [reply Engine/src/Engine/ECS/SparseArray.h:15]: In the future, this class will likely be improved upon, especially in terms of memory allocation. So thats why it is there

File: Engine/src/Engine/ECS/System.h
Comment (GuanHuiLim) [reply Engine/src/Engine/ECS/System.h]: It has been wrapped

File: Engine/src/Engine/ECS/SystemManager.h
Comment (GuanHuiLim) [reply Engine/src/Engine/ECS/SystemManager.h]: It has been wrapped

File: Engine/src/Engine/ECS/SparseArray.h
Comment (spironan) [reply Engine/src/Engine/ECS/SparseArray.h:15]: ah ok

File: Engine/src/Engine/ECS/ECS.h
Comment (spironan) [reply Engine/src/Engine/ECS/ECS.h:4]: ohhh ok then just put a note that we just need to include this file

**Review by spironan — APPROVED** [review]
ok looks good besides the lack of comments

**Pull request merged**
